prody/atomic/atomgroup.py

```diff
@@ -303,15 +312,29 @@ def __add__(self, other):
                     that = np.zeros(shape, this.dtype)
                 new._setFlags(key, np.concatenate((this, that)))
 
-        new._n_atoms = self._n_atoms + other._n_atoms
+        if newAG:
+            new._n_atoms = self._n_atoms + other._n_atoms
```

Comment:
this could go after line 244 to not add another if block

Reply:
Moving this line to 244 would affect len(self) used at line 306 resulting in a wrong array size on line 312

I tried it with the following code:

import prody

ag1 = prody.parsePDB('taxol.pdb') # small molecule in pdb format for flags mismatch
ag2 = prody.parsePDB('prody/tests/datafiles/pdb6flr.pdb')

# len(ag1) -> 62
# len(ag2) -> 6073
ag1.extend(ag2)

at line 314: new._setFlags(key, np.concatenate((this, that)))
np.concatenate((this, that)) makse anrray of 12208 for key 'ca'

While no execption is raise by _setFlags()

len(ag1._flags['hetatm']) ->  6135 which is correct
len(ag1._flags['ca']) -> 18281 which is incorrect

Reply:
Ok, thanks

Reply:
actually the bug persists in my original code and I found that the problem comes from no considering aliases for flags. For instance 'ca' an 'calpha' are aliases. While looping over the flags the code hit 'ca' and extended the array for the flag to be new size.  When it hits 'calpha' it gets the new size for  self and extends it again with the size of other.
I added some code now to handle aliases when concatenating flags in __add__. Is it sufficient for me to push the change in the branch in my forked repo ?

Reply:
Yes, pushing to the same branch will update the PR